opencensus/stats/internal/delta_producer.cc

```diff
@@ -0,0 +1,154 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "opencensus/stats/internal/delta_producer.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "absl/synchronization/mutex.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
+#include "opencensus/stats/bucket_boundaries.h"
+#include "opencensus/stats/internal/measure_data.h"
+#include "opencensus/stats/internal/measure_registry_impl.h"
+#include "opencensus/stats/internal/stats_manager.h"
+
+namespace opencensus {
+namespace stats {
+
+void Delta::Record(std::initializer_list<Measurement> measurements,
+                   TagSet tags) {
+  auto it = delta_.find(tags);
+  if (it == delta_.end()) {
+    it = delta_.emplace_hint(it, std::piecewise_construct,
+                             std::make_tuple(std::move(tags)),
```

Comment:
Why move() here?

Reply:
Because we do not need the tags again, and TagSet is relatively expensive to copy with the vector of strings.